File: 2.py
arr_name = [
    {"name": "Lisa"},
    {"name": "Bart"},
    {"name": "Maggie"},
    {"name": "Masha"}
]


def interestingMethod(arr=[]):
    endName = arr.pop()
    finalString = ', '.join(
        map(
            str, (name.get('name') for name in arr)))
    return endName.get('name')\
        if len(arr) == 0\
        else finalString.removesuffix(', ')+'&'+endName.get('name')


print(interestingMethod(arr_name))

File: 2.js
arrName = [
  { name: 'Lisa' },
  { name: 'Bart' },
  { name: 'Maggie' },
  { name: 'Masha' },
];

function interestingMethod(arr) {
  let names = arr.map((i) => i.name);
  names.length > 1
    ? (names = names.slice(0, -1).join(', ') + '&' + names.slice(-1))
    : (names = names.join(''));
  return names;
}

console.log(interestingMethod(arrName));
